src/planet-a/SubdivSkeleton.h

```diff
@@ -0,0 +1,392 @@
+                  "ID_T must be an integral type, 4 bytes or less in size");
+
+    using combination_t = uint64_t;
+
+public:
+
+    using base_t::capacity;
+    using base_t::size;
+
+    ID_T create_root()
+    {
+        ID_T const id = base_t::create();
+        m_idChildCount.resize(capacity());
+        m_idChildCount[size_t(id)] = 0;
+        return id;
+    };
+
+    std::pair<ID_T, bool> create_or_get(ID_T a, ID_T b)
+    {
+        combination_t const combination = hash_id_combination(a, b);
+
+        // Try emplacing a blank element under this combination of IDs, or get
+        // existing element
+        auto const& [it, success] = m_parentsToId.try_emplace(combination, 0);
+
+        if (success)
+        {
+            // The space was free, and a new element was succesfully emplaced
+
+            // Create a new ID for real, replacing the blank one from before
+            it->second = id_int_t(create_root());
+
+            // Keep track of the new ID's parents
+            m_idToParents.resize(capacity());
+            m_idToParents[it->second] = combination;
+
+            // Increase child count of the two parents
+            m_idChildCount[size_t(a)] ++;
+            m_idChildCount[size_t(b)] ++;
+        }
+
+        return { ID_T(it->second), success };
+    }
+
+    ID_T get(ID_T a, ID_T b) const;
+
+    std::pair<ID_T, ID_T> get_parents(ID_T a);
+
+    void reserve(size_t n)
+    {
+        base_t::reserve(n);
+        m_idToParents.reserve(base_t::capacity());
+        m_idChildCount.reserve(base_t::capacity());
+    }
+
+    static constexpr combination_t
+    hash_id_combination(id_int_t a, id_int_t b) noexcept
+    {
+        id_int_t const ls = (a > b) ? a : b;
+        id_int_t const ms = (a > b) ? b : a;
+
+        return combination_t(ls)
+                | (combination_t(ms) << (sizeof(id_int_t) * 8));
+    }
+
+    static constexpr combination_t
+    hash_id_combination(ID_T a, ID_T b) noexcept
+    {
+        return hash_id_combination( id_int_t(a), id_int_t(b) );
+    }
+
+private:
+    std::unordered_map<combination_t, id_int_t> m_parentsToId;
+    std::vector<combination_t> m_idToParents;
+    std::vector<uint8_t> m_idChildCount;
+
+}; // class SubdivTree
+
+//-----------------------------------------------------------------------------
+
+enum class SkVrtxId : uint32_t {};
+
+using SkVrtxStorage_t = osp::IdRefCount<SkVrtxId>::Storage_t;
+
+/**
+ * @brief Uses a SubdivIdTree to manage relationships between Vertex IDs, and
+ *        adds reference counting features.
+ *
+ * This class does NOT store vertex data like positions and normals.
+ */
+class SubdivSkeleton
+{
+
+public:
+
+
+    SkVrtxId vrtx_create_root()
+    {
+        SkVrtxId const vrtxId = m_vrtxIdTree.create_root();
+        return vrtxId;
+    };
+
+    SkVrtxId vrtx_create_or_get_child(SkVrtxId a, SkVrtxId b)
+    {
+        auto const [vrtxId, created] = m_vrtxIdTree.create_or_get(a, b);
+        return vrtxId;
```

Comment:
wouldn't you want the caller to know if the child was created or not?

Reply:
It was used before, but as changes came along, it stopped being used. There is no code that uses the 'created' return value, might as well remove it.